experimental/iterators/lib/Dialects/Iterators/IR/Iterators.cpp

```diff
@@ -31,13 +32,98 @@ void IteratorsDialect::initialize() {
       >();
 }
 
+//===----------------------------------------------------------------------===//
+// Iterators interfaces
+//===----------------------------------------------------------------------===//
+
+#include "iterators/Dialect/Iterators/IR/IteratorsOpInterfaces.cpp.inc"
+#include "iterators/Dialect/Iterators/IR/IteratorsTypeInterfaces.cpp.inc"
+
 //===----------------------------------------------------------------------===//
 // Iterators operations
 //===----------------------------------------------------------------------===//
 
 #define GET_OP_CLASSES
 #include "iterators/Dialect/Iterators/IR/IteratorsOps.cpp.inc"
 
+LogicalResult OpenOp::verify() {
```

Comment:
Some of these verifiers could be removed if you adopted standard Tablegen predicates on types (you may need to create your own CPred-based stuff).
The way we go for such cases is to test 1 case and then all the rest are considered auto-generated and now correct by construction.

OTOH we also need to provision for the future; if you feel the verification rules are going to change in the future and will require visting the type when it becomes nested then this is fine for now and until we invent the generic rules that we will want to apply.

Reply:
I have left the test cases for now. We can remove those that check off-the-shelf behavior before merging.